md/multi-hop-locks.md

```diff
@@ -11,14 +11,17 @@ More importantly, they allow [payment decorrelation](https://medium.com/@rusty_l
 Correlation attacks are especially problematic if the first and last intermediate hops are colluding because they would learn source and destination of a payment.
 In addition, scriptless script multi-hop locks enable improved proof of payment and atomic multi path payments (see below).
 
-
 Notation
 ---
-- `nonce(i, m)` is the public nonce of user `i` for a MuSig signature on `m`  (note that we don't call nonces `R` here to avoid confusion with the right lock `R`).
+
+- `Xi := xi*G` is the public key of user `i`.
+- `nonce(i,m) := ki*G` is the public nonce of user `i` for a MuSig signature on `m` (note that we don't call nonces `R` here to avoid confusion with the right lock `R`).
+- `T := t*G` is an arbitrary tweak applied to the shared nonce.
 - `psig(i,m,T) := ki + H(nonce(i,m)+nonce(j,m)+T,m)*xi` is a partial 2-of-2 MuSig from user `i` with user `j` for `m`.
-- `adaptor_sig(i,m,T) := psig(i,m,t*G) + t`
-- `sig(m,T) = psig(i,m,T) + adaptor_sig(j,m,T)` is the completed 2-of-2 MuSig from user i and j. It can be computed from a partial signature and an adaptor signature.
+- `adaptor_sig(i,m,T) := psig(i,m,T) + t`
+- `sig(m,T) := psig(i,m,T) + adaptor_sig(j,m,T)` is the completed 2-of-2 MuSig from user i and j. It can be computed from a partial signature and an adaptor signature.
 
+Note that the MuSig aggregated public key of `Xi` and `Xj` is ommitted from the equations for simplicity.
```

Comment:
I think it would make sense to just add the public key to the equation. As far as I can see only the definition of psig would be affected.
Perhaps something like
```
- `combine(Xi, Xj)` is the MuSig-combined public key of keys `Xi` and `Xj`.
- `psig(i,m,T) := ki + H(nonce(i,m)+nonce(j,m)+T, combine(Xi, Xj), m)*xi` is a partial 2-of-2 MuSig from user `i` with user `j` for `m`.
```

Reply:
Sounds good, I agree that it's clearer to have the complete, accurate equation.
I'll update that.

Reply:
If we really want to be accurate that's not exactly it.
I'll do the full version, let me know if I misunderstood something or if you think it makes it too verbose.
The order of R and X in the hash is different in bip-schnorr and in the MuSig paper: I'll use the bip-schnorr order here.

Reply:
Hm you're right. Now that we define `xi` as being the secret key to `Xi` we can't just imply anymore that it's the already musig-tweaked secret key. I'm now remembering that I left out these details intentionally to keep this relatively simple. But if it confused even you we should change it.

Here's my suggestion:
* I don't think we need to define what `combine` does exactly because it's not really needed for this writeup. Same for encode.
* We can say that `Xi` is the public key of user `i` but without mentioning `xi`.
* In combine we say that `combine(Xi, Xj) = (xi + xj)*G` and thereby we define xi and xj. Perhaps mention that xi and xj are the MuSig-adjusted secret keys from i and j?
* We remove the `H(encode(Xi,Xj),Xi)` from the definition of `psig`

What do you think?

Reply:
> I don't think we need to define what combine does exactly because it's not really needed for this writeup. Same for encode.

I agree, and it's indeed a bit too heavy to include all these details whereas the schemes below work regardless of how `encode` and `combine` are instantiated.

I'll re-work this with your suggestions.

Reply:
I've simplified this in the last version by just defining the combined public key, let me know what you think!